### PR TITLE
feat(jans-config-api): user-management endpoints

### PR DESCRIPTION
**Pull request by pujavs**
@yuriyz request you to pls review and approve new endpoints for user management
Related issue [418](https://github.com/JanssenProject/jans/issues/418)

**Endpoint:** /jans-config-api/api/v1/user

### Impacted modules:
**1) jans-config-api:**  Endpoint related classes and swagger spec
**2)jans-cli:** endpoint spec changes

### CODE REVIEW

File: jans-config-api/server/src/main/java/io/jans/configapi/rest/resource/auth/UserResource.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            logger.debug("User serach by inum:{}", escapeLog(inum));
+        }
+        User user = userSrv.getUserByInum(inum);
+        logger.debug("user:{}", user);
+        return Response.ok(user).build();
+    }
+
+    @POST
+    @ProtectedApi(scopes = { ApiAccessConstants.USER_WRITE_ACCESS })
+    public Response createOpenIdConnect(@Valid User user) {
```
Comment (yuriyz) [inline jans-config-api/server/src/main/java/io/jans/configapi/rest/resource/auth/UserResource.java]: Not sure what does method name mean `createOpenIdConnect`, sounds like typo. `createUser` fits better.

File: jans-config-api/server/src/main/java/io/jans/configapi/rest/resource/auth/UserResource.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+    @Inject
+    Logger logger;
+
+    @Inject
+    UsrService userSrv;
+
+    @GET
+    @ProtectedApi(scopes = { ApiAccessConstants.USER_READ_ACCESS })
+    public Response getOpenIdConnectClients(
```
Comment (yuriyz) [inline jans-config-api/server/src/main/java/io/jans/configapi/rest/resource/auth/UserResource.java]: It's not openid clients, we are returning users. `searchUsers` or `getUsers`

File: jans-config-api/server/src/main/java/io/jans/configapi/rest/resource/auth/UserResource.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            @DefaultValue(DEFAULT_LIST_SIZE) @QueryParam(value = ApiConstants.LIMIT) int limit,
+            @DefaultValue("") @QueryParam(value = ApiConstants.PATTERN) String pattern,
+            @DefaultValue(DEFAULT_LIST_START_INDEX) @QueryParam(value = ApiConstants.START_INDEX) int startIndex,
+            @QueryParam(value = ApiConstants.SORT_BY) String sortBy,
+            @QueryParam(value = ApiConstants.SORT_ORDER) String sortOrder) {
+        if (logger.isDebugEnabled()) {
+            logger.debug("User serach param - limit:{}, pattern:{}, startIndex:{}, sortBy:{}, sortOrder:{}",
```
Comment (yuriyz) [inline jans-config-api/server/src/main/java/io/jans/configapi/rest/resource/auth/UserResource.java]: Typo, `serach` -> `search`

**Review by yuriyz — CHANGES_REQUESTED** [review]
Some minor changes are needed, please check comments.

File: jans-config-api/server/src/main/java/io/jans/configapi/service/auth/UsrService.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+
+@ApplicationScoped
+@Named("userSrv")
+public class UsrService extends UserService {
```
Comment (yuriyz) [inline jans-config-api/server/src/main/java/io/jans/configapi/service/auth/UsrService.java]: `public class UserService extends io.jans.as.common.service.common.UserService` will work too.

Comment (pujavs) [reply jans-config-api/server/src/main/java/io/jans/configapi/service/auth/UsrService.java]: it was `public class UserService extends io.jans.as.common.service.common.UserService` but when analyzed it locally it was giving sonar issue to change the name and hence i changed it to ` UsrService`.
Will check in as suggested and lets see

Comment (yuriyz) [reply jans-config-api/server/src/main/java/io/jans/configapi/service/auth/UsrService.java]: I see, let it be then.

File: jans-config-api/server/src/main/java/io/jans/configapi/rest/resource/auth/UserResource.java (hunk 1 above)
Comment (pujavs) [reply jans-config-api/server/src/main/java/io/jans/configapi/rest/resource/auth/UserResource.java]: Sorry, rectified the same

File: jans-config-api/server/src/main/java/io/jans/configapi/rest/resource/auth/UserResource.java (hunk 2 above)
Comment (pujavs) [reply jans-config-api/server/src/main/java/io/jans/configapi/rest/resource/auth/UserResource.java]: Sorry, rectified the same

File: jans-config-api/server/src/main/java/io/jans/configapi/rest/resource/auth/UserResource.java (hunk 3 above)
Comment (pujavs) [reply jans-config-api/server/src/main/java/io/jans/configapi/rest/resource/auth/UserResource.java]: rectified

File: jans-config-api/server/src/main/java/io/jans/configapi/service/auth/UsrService.java (hunk 4 above)
Comment (pujavs) [reply jans-config-api/server/src/main/java/io/jans/configapi/service/auth/UsrService.java]: Strangely it is giving error locally but not reported by Github sonar check
image

**Review by yuriyz — DISMISSED** [review]

**Review by mbaser — APPROVED** [review]

**Review by yuriyz — APPROVED** [review]

**Pull request merged**